pkg/apis/core/types_shoot.go

```diff
@@ -539,6 +539,11 @@ type KubeProxyConfig struct {
 	// Mode specifies which proxy mode to use.
 	// defaults to IPTables.
 	Mode *ProxyMode
+	// Enabled indicates whether kube-proxy should be deployed or not.
+	// Depending on your networking provider extensions switching kube-proxy off might be rejected.
+	// For example, cilium supports running without kube-proxy, but calico usually needs kube-proxy except for its ebpf dataplane.
```

Comment:
Sorry, but please remove those specific information again from the core API. The sentence above is good enough :) Maybe with the addition that consulting the documentation fo the used network extension is recommended.

Reply:
Just to be clear (as running make generate tends to overheat/crash my system and I want to reduce to the number of reboots): Should the documentation comment read `// Enabled...` + `// Depending...` or go back to just being `// Enabled...`?

Reply:
Proposal:

```golang
// Enabled indicates whether kube-proxy should be deployed or not.
// Depending on your networking extensions switching kube-proxy off might be rejected. Consulting the respective documentation of the used networking extension is recommended before using this field.
```

Reply:
Adapted the documentation to reflect almost completely to the proposal (hint: one word changed) in the new commit.